pkg/client/client_test.go

```diff
@@ -952,6 +953,109 @@ func TestCreateOrUpdateService(t *testing.T) {
 	}
 }
 
+func TestCreateOrUpdateServiceAccount(t *testing.T) {
+	ctx := context.Background()
+	testCases := []struct {
```

Comment:
could you add a test ensuring that no update is triggered when the metadata doesn't change but the SA secrets have changed?